nautilus_core/network/src/websocket.rs

```diff
@@ -359,6 +364,7 @@ impl WebSocketClient {
     #[staticmethod]
     fn connect(
         url: String,
+        headers:Vec<(&str,&str)>,
```

Comment:
You'll need to take a `Vec<(String, String)>` because having lifetimes will be problematic in the Python API. And you want to store these headers in the `WebsocketInnerClient` so that they can be re-used in case of a reconnect.